torch/csrc/jit/passes/quantization.cpp

```diff
@@ -575,51 +575,48 @@ void InsertQuantDeQuantImpl(
     }
   }
 
-  std::vector<Value*> values_to_quantize;
-  std::unordered_map<script::ModulePtr, script::Module>
-      child_modules_to_quantize;
   QuantizeHelper qh(module);
   std::stack<Block*> blocks_to_visit;
   blocks_to_visit.push(graph->block());
   while (!blocks_to_visit.empty()) {
     Block* b = blocks_to_visit.top();
     blocks_to_visit.pop();
-    for (Node* n : b->nodes()) {
+    for (auto it = b->nodes().begin(), end=b->nodes().end(); it != end;) {
+      Node* n = *it++;
       for (Value* v : n->outputs()) {
         if (v->type()->isSubtypeOf(TensorType::get())) {
-          auto child_module = qh.findChildModuleToQuantize(v);
-          if (child_module) {
-            child_modules_to_quantize[child_module.value().module_object()] =
-                child_module.value();
+          if (v->node()->kind() == prim::CallMethod) {
+            auto module_instance = v->node()->inputs()[0];
+            auto module_method_name = v->node()->s(attr::name);
+            // calling method on self
+            if (module_instance == graph->inputs()[0]) {
+              InsertQuantDeQuantImpl(module, module_method_name);
+            } else {
+              auto child_module = qh.findChildModuleToQuantize(v);
```

Comment:
I think this place can be cleaned up a bit.

For starters, we don't need `if (v->node()->kind() == prim::CallMethod) {` check in `findChildModuleToQuantize` anymore.

Next, I think the code above is very similar to the code inside that function, so I think it would be good if we can just reuse it. The only difference is that in one case we check `module_instance == graph->inputs()[0]` and in the other we're looking at submodules. If that's refactoring doesn't work out, I think it would still make sense to factor out the code above to a separate function, so that we're operating on the same level of details in a scope of one function (right now the code above is one level lower than a call to `findChildModuleToQuantize` which follow that code).

Reply:
Yeah I think there are some duplicated code to get module_instance and module_method_name, I can remove these in findChildModuleToQuantize function

Reply:
I think in both cases we need to know module_instance and module_method_name, so it's not different level of details, just in the case of child modules we also need to get the child module, and in case of self, the module is already there so we don't need to call another function to get it.